assets/src/vo/date-time/datetime.js

```diff
@@ -0,0 +1,1156 @@
+	 * @param {number} offset  In minutes unless > -16 or < -16 in which case it
+	 * is treated as hours.
+	 * @param {string} locale
+	 * @return {DateTime}  An instance of DateTime
+	 */
+	static fromISOWithOffset(
+		ISOString,
+		offset = DEFAULT_OFFSET,
+		locale = DEFAULT_VALID_LOCALE
+	) {
+		DateTime.assertISO8601IsValid( ISOString );
+		DateTime.assertIsOffset( offset );
+		DateTime.assertLocaleIsValid( locale );
+		const datetime = isEmpty( ISOString ) ?
+			moment().utcOffset( offset, true ).locale( locale ) :
+			moment.utc( ISOString )
+				.utcOffset( offset, true )
+				.locale( locale );
+		return new DateTime( datetime );
+	}
+
+	/**
+	 * Constructs a DateTime from a javascript Date object.
+	 *
+	 * @param {Date} date
+	 * @param {string} timezone
+	 * @param {string} locale
+	 * @return {DateTime} Returns an instance of DateTime
+	 */
+	static fromJSDate(
+		date,
+		timezone = DEFAULT_TIMEZONE_STRING,
+		locale = DEFAULT_VALID_LOCALE
+	) {
+		DateTime.assertIsDate( date );
+		DateTime.assertTimezoneIsValid( timezone );
+		DateTime.assertLocaleIsValid( locale );
+		return new DateTime(
+			moment( date ).tz( timezone ).locale( locale )
+		);
+	}
+
+	/**
+	 * Constructs a Datetime from a javascript Date object.
+	 *
+	 * The difference between this and fromJSDate is that this can be set with
+	 * an offset vs a timezone string.
+	 *
+	 * @param {Date} date
+	 * @param {number} offset
+	 * @param {string} locale
+	 * @return {DateTime} Returns an instance of DateTime
+	 */
+	static fromJSDateWithOffset(
+		date,
+		offset = DEFAULT_OFFSET,
+		locale = DEFAULT_VALID_LOCALE
+	) {
+		DateTime.assertIsDate( date );
+		DateTime.assertIsOffset( offset );
+		DateTime.assertLocaleIsValid( locale );
+		return new DateTime(
+			moment( date ).utcOffset( offset ).locale( locale )
+		);
+	}
+
+	/**
+	 * Constructs a DateTime (in utc) with milliseconds from epoch.
+	 *
+	 * @param {number} milliseconds
+	 * @param {string} locale
+	 * @return {DateTime} Returns an instance of DateTime
+	 * @throws TypeError
+	 */
+	static fromMillis( milliseconds, locale = DEFAULT_VALID_LOCALE ) {
+		DateTime.assertLocaleIsValid( locale );
+		if ( ! isNumber( milliseconds ) ) {
+			throw new TypeError( 'Provided value must be a number' );
+		}
+		return new DateTime(
+			moment.utc( milliseconds ).locale( locale )
+		);
+	}
+
+	/**
+	 * Constructs a DateTime in utc() with seconds from epoch.
+	 *
+	 * @param {number} seconds
+	 * @param {string} locale
+	 * @return {DateTime} An instance of DateTime
+	 * @throws TypeError
+	 */
+	static fromUnix( seconds, locale = DEFAULT_VALID_LOCALE ) {
+		DateTime.assertLocaleIsValid( locale );
+		if ( ! isNumber( seconds ) ) {
+			throw new TypeError( 'Provided value must be a number' );
+		}
+		return new DateTime(
+			moment.unix( seconds ).utc().locale( locale )
+		);
+	}
+
+	/**
+	 * Constructs a DateTime from an object of values assuming its in "local"
+	 * time (browser or server if run server side).
```

Comment:
Sorry, I didn't change this because I don't think what you suggested is any clearer.

Reply:
... assumes it is in "local" time (browser or server) :)

To elaborate:  meaning if the app using this library is being viewed in a browser, then the object of values passed in is considered to be in the local time as set for the browser.  However, it's also possible that the app using the value object is running in the context of a server (i.e. node server side rendering, or unit tests) in which case the "local" time would be derived from the server not the browser.

Pretty sure you understood it correctly originally.  I was just pointing out that I thought what you suggested as a change to make it _more_ clear was redundant (and didn't really make it any more clear than it already was).  There are times when you suggest changes to jsdoc text and I'm completely on board with it because it makes sense and its definitely clearer than what was originally there and there are times where I think its unnecessarily redundant and doesn't really improve on what was already there (those times are few but this is one of them).

It appears to me (based on your original suggested change) that you _understood_ it correctly and your first impressions of meaning was correct.  So why are we bike-shedding over this?